sys/net/network_layer/ng_sixlowpan/frag/ng_sixlowpan_frag.c

```diff
@@ -80,18 +80,19 @@ static uint16_t _send_1st_fragment(ng_sixlowpan_netif_t *iface, ng_pktsnip_t *pk
                                    size_t payload_len, size_t datagram_size)
 {
     ng_pktsnip_t *frag;
-    uint16_t max_frag_size = _floor8(iface->max_frag_size -
-                                     (payload_len - datagram_size) -
-                                     sizeof(ng_sixlowpan_frag_t));
     uint16_t local_offset = 0;
+    /* payload_len: actual size of the packet vs
+     * datagram_size: size of the uncompressed IPv6 packet */
+    int payload_diff = (datagram_size - payload_len);
```

Comment:
I've chosen the pseudo code `len(iphc_dispatch)` to make clear, that though this is the dispatch length, it is different from what `dispatch_len` was before (which only applied to non-IPHC dispatches ;-)).